src/analysis/Solution.cpp

```diff
@@ -129,6 +132,11 @@ void Solution::calc_analytical_solution(real t) {
     if (m_current_time_step == t) {
         return;
     }
+
+    if (!m_has_analytical_solution) {
```

Comment:
should we combine it with the inquiry before?

Reply:
i dont think so. Each query has its own semantic purpose. The first one checking it the simulation finished the second if an analytical solution exists. The compiler should optimize this if it makes a difference. I my opinion this is not worth mixing up code.

Reply:
ok